File: FlightReservation/src/main/java/com/sarath/flightreservation/entities/Flight.java
package com.sarath.flightreservation.entities;

import java.sql.Timestamp;
import java.util.Date;

import com.fasterxml.jackson.annotation.JsonFormat;

import jakarta.persistence.Entity;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;


@EqualsAndHashCode(callSuper = true)
@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Flight extends AbstractEntity
{
    private String flight_number;
    private String operating_airlines;
    private String departure_city;
    private String arrival_city;
    @JsonFormat(pattern = "MM-dd-yyyy")
    private Date date_of_departure;
    private Timestamp estimate_departure_time;
}

File: FlightAngularView/src/app/interfaceHybrid/Flight.ts
import { Time } from "@angular/common";

export interface Flight
{
    id:number,
    flight_number:string,
    operating_airlines:string,
    departure_city:string,
    arrival_city:string,
    date_of_departure:Date,
    estimate_departure_time:Time

}

export class FlightClass implements Flight {
    constructor(
        public id:number,
        public flight_number:string,
        public operating_airlines:string,
        public departure_city:string,
        public arrival_city:string,
        public date_of_departure:Date,
        public estimate_departure_time:Time
    ) {}
  
    // Add behavior/methods if needed
    public calculateDuration(): number {
      // Implementation
      return 0;
    }
  }